flink-java/src/main/java/org/apache/flink/api/java/functions/IdentityKeySelector.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.functions;
```

Comment:
A better solution than using a KeySelector would be to extend the Keys.ExpressionKey class to support AtomicTypes. The code from `checkForAtomicType()` could be moved to the constructor of `Keys.ExpressionKey`. The `Keys.ExpressionKey.keyFields` member should be a list with a single `FlatFieldDescriptor` that points to flat position `0` and have the type of the AtomicType.

Generalizing the ExpressionKey to AtomicType should not have side effects, but you should run a build with all tests to verify that.